golem-common/src/model/mod.rs

```diff
@@ -2482,6 +2483,380 @@ impl FromStr for ComponentType {
     }
 }
 
+/// Key that can be used to identify a component file.
+/// All files with the same content will have the same key.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize, NewType)]
+pub struct InitialComponentFileKey(pub String);
+
+impl Display for InitialComponentFileKey {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+
+/// Path inside a component filesystem. Must be
+/// - absolute (start with '/')
+/// - not contain ".." components
+/// - not contain "." components
+/// - use '/' as a separator
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub struct ComponentFilePath(Utf8UnixPathBuf);
```

Comment:
As this guy is getting serialized and we might have windows binaries at some point I went with this instead of a normal path.